platform/utm_mgrs_utils.cpp

```diff
@@ -0,0 +1,567 @@
+ * additional northings. You have to figure out how many 2000000 meters need
+ * to be added for the zone letter of the MGRS coordinate.
+ *
+ * @param n
+ *            second letter of the MGRS 100k zone
+ * @param set
+ *            the MGRS table set number, which is dependent on the UTM zone
+ *            number.
+ */
+double square_char_to_northing(char n, int set) {
+
+    if (n > 'V')
+        return -1;
+
+    int curRow = SET_ORIGIN_ROW_LETTERS[set - 1];
+    double northingValue = 0.0;
+    bool rewindMarker = false;
+
+    while (curRow != n)
+    {
+        curRow++;
+        if (curRow == 'I')
+            curRow++;
+        if (curRow == 'O')
+            curRow++;
+        // fixing a bug making whole application hang in this loop
+        // when 'n' is a wrong character
+        if (curRow > 'V') {
+            if (rewindMarker) // making sure that this loop ends
+                return -1;
+            curRow = 'A';
+            rewindMarker = true;
+        }
+        northingValue += 100000.0;
+    }
+
+    return northingValue;
+}
+
+/**
+ * The function getMinNorthing returns the minimum northing value of a MGRS
+ * zone.
+ *
+ * portted from Geotrans' c Latitude_Band_Value structure table. zoneLetter
+ * : MGRS zone (input)
+ */
+double zone_to_min_northing(char zoneLetter)
+{
+    double northing;
+    switch (zoneLetter)
+    {
+        case 'C':
+            northing = 1100000.0;
+            break;
+        case 'D':
+            northing = 2000000.0;
+            break;
+        case 'E':
+            northing = 2800000.0;
+            break;
+        case 'F':
+            northing = 3700000.0;
+            break;
+        case 'G':
+            northing = 4600000.0;
+            break;
+        case 'H':
+            northing = 5500000.0;
+            break;
+        case 'J':
+            northing = 6400000.0;
+            break;
+        case 'K':
+            northing = 7300000.0;
+            break;
+        case 'L':
+            northing = 8200000.0;
+            break;
+        case 'M':
+            northing = 9100000.0;
+            break;
+        case 'N':
+            northing = 0.0;
+            break;
+        case 'P':
+            northing = 800000.0;
+            break;
+        case 'Q':
+            northing = 1700000.0;
+            break;
+        case 'R':
+            northing = 2600000.0;
+            break;
+        case 'S':
+            northing = 3500000.0;
+            break;
+        case 'T':
+            northing = 4400000.0;
+            break;
+        case 'U':
+            northing = 5300000.0;
+            break;
+        case 'V':
+            northing = 6200000.0;
+            break;
+        case 'W':
+            northing = 7000000.0;
+            break;
+        case 'X':
+            northing = 7900000.0;
+            break;
+        default:
+            northing = -1.0;
+    }
+
+    return northing;
+}
```

Comment:
Well, ok here, but please setup your editor:
- use 2 spaces for _tab_ instead of 4
- use C++ braces, from the new line, not K&R style
- do not use braces in _one line_ expressions

Reply:
Reformatted code.